src/test/java/com/hillel/items_exchange/util/PatternHandlerTest.java

```diff
@@ -0,0 +1,307 @@
+                "~~``~~",
+                "`_'",
+                "'@'\"`!",
+                "1",
+                "$a",
+                "$",
+                "a$",
+                "$+a",
+                "a+$"));
+
+        correctNames.addAll(getStringsWithSpecialSymbol("pu", "shkin", Collections.emptyList()));
+        correctNames.addAll(getStringsWithRussianOrUkrCharacter("aaa", "bbb"));
+
+        return correctNames;
+    }
+
+    private static List<String> createWrongUserNames() {
+
+        return List.of(
+                "",
+                " ",
+                "aaa bbb",
+                " aaa",
+                "aaa ");
+    }
+
+    private static List<String> createCorrectWordsEmptyOrMin2Max50() {
+        List<String> correctWords = new ArrayList<>(List.of(
+                "",
+                "aA",
+                "11",
+                "aA1",
+                "aA1-",
+                "aA1-a",
+                "aA1'",
+                "aA1'a",
+                "aA1`",
+                "aA1`a",
+                "aA1_",
+                "aA1_a",
+                "LengthIs50LengthIs50LengthIs50LengthIs50LengthIs50"));
+
+        correctWords.addAll(getStringsWithRussianOrUkrCharacter("aa", "bb"));
+
+        return correctWords;
+    }
+
+    private static List<String> createWrongWordsEmptyOrMin2Max50() {
+        List<String> wrongWords = new ArrayList<>(List.of(
+                "w",
+                "W",
+                "1",
+                " ",
+                " 1",
+                " W",
+                " w",
+                "1 ",
+                "1 ",
+                "W ",
+                "w ",
+                "ThisWordHasLengthMoreThan50CharactersItsLengthIs51S"));
+
+        List<Character> correctCharacters = List.of('\'', '_', '`', '-');
+        wrongWords.addAll(getStringsWithSpecialSymbol("aaa", "bbb", correctCharacters));
+        wrongWords.addAll(getStringsWithSpecialSymbol("", "", Collections.emptyList()));
+        wrongWords.addAll(getStringsWithRussianOrUkrCharacter("", ""));
+
+        return wrongWords;
+    }
+
+    private static List<String> getStringsWithSpecialSymbol(String str1, String str2, List<Character> deletedCharacters) {
+        List<Character> addedCharacters = new ArrayList<>(List.of(
+                '`', '~', '!', '@', '#', '$', '%', '^', '&', '*', '(', ')', '+', '=', '[', ']', '{', '}', '\'', '"',
+                ':', ';', '/', '\\', '|', '?', '.', ',', '<', '>', '-', '_'));
+        addedCharacters.removeAll(deletedCharacters);
+
+        return addedCharacters.stream()
+                .map(character -> str1.concat(String.valueOf(character)).concat(str2))
+                .collect(Collectors.toList());
+    }
+
+    private static List<String> getStringsWithRussianOrUkrCharacter(String str1, String str2) {
+        List<Character> russianChars = List.of(
```

Comment:
As I can see these chars are not only russian.

Reply:
It was until you changed pattern))

Reply:
Make it with using ASCII codes (https://www.ascii-codes.com/cp855.html) and IntRange as well.